### PR TITLE
refactor(frontend): add data-test attribute for text input

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

There's no class-agnostic way of identifying text input group for our e2e test

## Solution

Add a div in between lib components for `data-test`

## Tests

- [x] All text input field visuals don't get affected
  - [x] plain text
  - [x] rich text
  - [x] text input with trash can

## Deploy Notes

N/A

### CODE REVIEW

File: packages/frontend/src/components/RichTextEditor/index.tsx [hunk 1]
```diff
@@ -199,28 +199,30 @@ const RichTextEditor = ({
...
-        render={({ field: { onChange, value } }) => (
-          <Editor
-            onChange={onChange}
-            initialValue={value}
-            editable={!disabled}
-            placeholder={placeholder}
-            variablesEnabled={variablesEnabled}
-            isRich={isRich}
-          />
+      <div data-test="text-input-group">
```
Comment (m0nggh) [inline packages/frontend/src/components/RichTextEditor/index.tsx]: nit: use `Box` instead of div?

**Review by m0nggh — APPROVED** [review]
other than the nit, tested that the inputs still work, lgtm!

File: packages/frontend/src/components/RichTextEditor/index.tsx (hunk 1 above)
Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/index.tsx]: Cannot use `Box` or element from frameworks cos they don't expose `data-test` attrib

Comment (m0nggh) [reply packages/frontend/src/components/RichTextEditor/index.tsx]: oh wait i thought u did that for the `FlowStepHeader` in this [PR](https://github.com/opengovsg/plumber/pull/450/files)?

Comment (stanleynguyen) [reply packages/frontend/src/components/RichTextEditor/index.tsx]: Oh yea hor lemme double check whether I can do it right on the framework elem

**Pull request merged**